core/src/main/scala/cats/Reducible.scala

```diff
@@ -0,0 +1,183 @@
+package cats
+
+import simulacrum._
+
+import Fold.{Return, Pass, Continue}
+
+/**
+ * Data structures that can be reduced to a summary value.
+ *
+ * `Reducible` is like a non-empty `Foldable`. In addition to the fold
+ * methods it provides reduce methods which do not require an initial
+ * value.
+ *
+ * In addition to the methods needed by `Foldable`, `Reducible` is
+ * implemented in terms of two methods:
+ *
+ *  - `reduceLeftTo(fa)(f)(g)` eagerly reduces with an additional mapping function
+ *  - `reduceRightTo(fa)(f)(g)` lazily reduces with an additional mapping function
+ */
+@typeclass trait Reducible[F[_]] extends Foldable[F] { self =>
+
+  /**
+   * Left-associative reduction on 'F' using the function 'f'.
+   *
+   * Implementations should override this method when possible.
+   */
+  def reduceLeft[A](fa: F[A])(f: (A, A) => A): A =
+    reduceLeftTo(fa)(identity)(f)
+
+  /**
+   * Right-associative reduction on 'F' using the function 'f'.
+   */
+  def reduceRight[A](fa: F[A])(f: A => Fold[A]): Lazy[A] =
+    reduceRightTo(fa)(identity)(f)
+
+  /**
+   * Reduce a `F[A]` value using the given semigroup.
+   */
+  def reduce[A](fa: F[A])(implicit A: Semigroup[A]): A =
+    reduceLeft(fa)(A.combine)
+
+  /**
+   * Reduce a `F[G[A]]` value using `SemigroupK[G]`, a universal
+   * semigroup for `G[_]`.
+   *
+   * This method is a generalization of `reduce`.
+   */
+  def reduceK[G[_], A](fga: F[G[A]])(implicit G: SemigroupK[G]): G[A] =
+    reduce(fga)(G.algebra)
+
+  /**
+   * Apply f to each element of fa and combine them using the given
+   * Semigroup[B].
+   */
+  def reduceMap[A, B](fa: F[A])(f: A => B)(implicit B: Semigroup[B]): B =
+    reduceLeftTo(fa)(f)((b, a) => B.combine(b, f(a)))
+
+  /**
+   * Apply f to each element of fa and combine them with the given
+   * associative function g.
+   */
+  def reduceLeftTo[A, B](fa: F[A])(f: A => B)(g: (B, A) => B): B
+
+  /**
+   * Overriden from Foldable[_] for efficiency.
+   */
+  override def reduceLeftToOption[A, B](fa: F[A])(f: A => B)(g: (B, A) => B): Option[B] =
+    Some(reduceLeftTo(fa)(f)(g))
+
+  /**
+   * Apply f to each element of fa and lazily combine them with the
+   * given associative function `g`.
```

Comment:
Here and in `reduceLeftTo`, the ScalaDoc says "Apply f to each element", but wouldn't it only be applied to the initial element?